clients/unshaded/src/main/java/tachyon/client/block/BSContext.java

```diff
@@ -0,0 +1,207 @@
+        getWorkerAddress(NetworkAddressUtils.getLocalHostName(ClientContext.getConf()));
+
+    // If the local worker is not available, do not initialize the local worker client pool
+    if (null == localWorkerAddress) {
+      mLocalBlockWorkerClientPool = null;
+    } else {
+      mLocalBlockWorkerClientPool =
+          new BlockWorkerClientPool(localWorkerAddress);
+    }
+  }
+
+  /**
+   * Re-initializes the Block Store context. This method should only be used in ClientContext.
+   *
+   * TODO: Prevent classes other than ClientContext from accessing this method.
+   */
+  public void resetContext() {
+    mBlockMasterClientPool.close();
+    if (mLocalBlockWorkerClientPool != null) {
+      mLocalBlockWorkerClientPool.close();
+    }
+    mBlockMasterClientPool =
+        new BlockMasterClientPool(ClientContext.getMasterAddress());
+    NetAddress localWorkerAddress =
+        getWorkerAddress(NetworkAddressUtils.getLocalHostName(ClientContext.getConf()));
+
+    // If the local worker is not available, do not initialize the local worker client pool
+    if (null == localWorkerAddress) {
+      mLocalBlockWorkerClientPool = null;
+    } else {
+      mLocalBlockWorkerClientPool =
+          new BlockWorkerClientPool(localWorkerAddress);
+    }
+  }
+
+  private NetAddress getWorkerAddress(String hostname) {
+    BlockMasterClient masterClient = acquireMasterClient();
+    try {
+      List<WorkerInfo> workers = masterClient.getWorkerInfoList();
+      if (hostname.isEmpty()) {
+        // TODO: Do this in a more defined way
+        return workers.get(0).getAddress();
+      }
+      for (WorkerInfo worker : workers) {
+        if (worker.getAddress().getHost().equals(hostname)) {
+          return worker.getAddress();
+        }
+      }
+      return null;
+    } catch (Exception e) {
+      return null;
+    } finally {
+      releaseMasterClient(masterClient);
+    }
+  }
+
+  /**
+   * Acquires a block master client from the block master client pool.
+   *
+   * @return the acquired block master client
+   */
+  public BlockMasterClient acquireMasterClient() {
+    return mBlockMasterClientPool.acquire();
+  }
+
+  /**
+   * Releases a block master client into the block master client pool.
+   *
+   * @param masterClient a block master client to release
+   */
+  public void releaseMasterClient(BlockMasterClient masterClient) {
+    mBlockMasterClientPool.release(masterClient);
+  }
+
+  /**
+   * Obtains a worker client to a worker in the system. A local client is preferred to be returned
+   * but not guaranteed. The caller should use {@link WorkerClient#isLocal} to verify if the client
+   * is local before assuming so.
+   *
+   * @return a WorkerClient to a worker in the Tachyon system
+   */
+  public WorkerClient acquireWorkerClient() {
+    if (mLocalBlockWorkerClientPool != null) {
+      return mLocalBlockWorkerClientPool.acquire();
+    } else {
+      return acquireRemoteWorkerClient("");
```

Comment:
does the empty string ("") here mean any remote worker client? If yes, please comment on that.